e2e/test/iothub/AuthenticationWithTokenRefreshDisposalTests.cs

```diff
@@ -0,0 +1,252 @@
+            DeviceClient deviceClient2 = DeviceClient.Create(testDevice.IoTHubHostName, authenticationMethod, transport);
+            await deviceClient2.SendEventAsync(new Client.Message()).ConfigureAwait(false);
+            await deviceClient2.CloseAsync();
+            deviceClient2.Dispose();
+            Logger.Trace("Test with instance 2 completed");
+
+            authenticationMethod.Dispose();
+        }
+
+        private async Task ReuseAuthenticationMethod_MuxedDevices(Client.TransportType transport, int devicesCount)
+        {
+            IList<TestDevice> testDevices = new List<TestDevice>();
+            IList<DeviceClient> deviceClients = new List<DeviceClient>();
+            IList<AuthenticationWithTokenRefresh> authenticationMethods = new List<AuthenticationWithTokenRefresh>();
+            IList<AmqpConnectionStatusChange> amqpConnectionStatuses = new List<AmqpConnectionStatusChange>();
+
+            // Set up amqp transport settings to multiplex all device sessions over the same amqp connection.
+            var amqpTransportSettings = new AmqpTransportSettings(transport)
+            {
+                AmqpConnectionPoolSettings = new AmqpConnectionPoolSettings
+                {
+                    Pooling = true,
+                    MaxPoolSize = 1,
+                },
+            };
+
+            for (int i = 0; i < devicesCount; i++)
+            {
+                TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+                var authenticationMethod = new DeviceAuthenticationSasToken(testDevice.ConnectionString);
+
+                testDevices.Add(testDevice);
+                authenticationMethods.Add(authenticationMethod);
+            }
+
+            // Initialize the client instances, set the connection status change handler and open the connection.
+            for (int i = 0; i < devicesCount; i++)
+            {
+                DeviceClient deviceClient = DeviceClient.Create(testDevices[i].IoTHubHostName, authenticationMethods[i], new ITransportSettings[] { amqpTransportSettings });
+
+                var amqpConnectionStatusChange = new AmqpConnectionStatusChange(testDevices[i].Id, Logger);
+                deviceClient.SetConnectionStatusChangesHandler(amqpConnectionStatusChange.ConnectionStatusChangesHandler);
+                amqpConnectionStatuses.Add(amqpConnectionStatusChange);
+
+                await deviceClient.OpenAsync().ConfigureAwait(false);
+                deviceClients.Add(deviceClient);
+            }
+
+            // Close and dispose client instance 1.
+            // The closed client should report a status of "disabled" while the rest of them should be connected.
+
+            await deviceClients[0].CloseAsync().ConfigureAwait(false);
+            deviceClients[0].Dispose();
+
+            amqpConnectionStatuses[0].LastConnectionStatus.Should().Be(ConnectionStatus.Disabled);
+
```

Comment:
Could also add amqpConnectionStatuses[1].LastConnectionStatus.Should().Be(ConnectionStatus.Connected);

Reply:
we actually check the status of the entire list of client instances right below